src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/DedupJobSchedulerDAO.java

```diff
@@ -0,0 +1,141 @@
+package com.lithium.dbi.rdbi.recipes.scheduler;
+
+import com.lithium.dbi.rdbi.BindArg;
+import com.lithium.dbi.rdbi.BindKey;
+import com.lithium.dbi.rdbi.Mapper;
+import com.lithium.dbi.rdbi.Query;
+
+import java.util.List;
+
+/**
+ * Main difference between this DAO and the ExclusiveJobSchedulerDAO is that this DAO
+ * allows the same job to be in the running queue and ready queue at the same time.
+ *
+ * Also while reserving jobs, a job is only reserved if the job has no instances in the
+ * running queue.
+ */
+public interface DedupJobSchedulerDAO {
+    /**
+     * When scheduling a job make sure its not already in the ready queue. A job
+     * can be scheduled regardless of whether its in the running queue.
+     */
+    @Query(
+        "local readyJobScore = redis.call('ZSCORE', $readyQueue$, $jobStr$)\n" +
+        "local isPaused = redis.call('GET', $pausedTube$) \n" +
+        "if not readyJobScore and not isPaused then\n" +
+        "   redis.call('ZADD', $readyQueue$, $runInMillis$, $jobStr$)\n" +
+        "   return 1\n" +
+        "else\n" +
+        "   return 0\n" +
+        "end"
+    )
+    int scheduleJob(
+            @BindKey("readyQueue") String readyQueue,
+            @BindKey("pausedTube") String pausedTube,
+            @BindArg("jobStr") String job,
+            @BindArg("runInMillis") long runInMillis);
+
+    /**
+     * When reserving jobs reserve jobs that are currently in ready queue but not in the running queue.
+     * If the job is in running queue need to wait till the running instance of the job has completed before
+     * we allow another reservation of the job from the ready queue.
+     */
+    @Mapper(TimeJobInfoListMapper.class)
+    @Query(
+        "local reserved = {}\n" +
+        "local isPaused = redis.call('GET', $pausedTube$) \n" +
+        "if isPaused then\n" +
+        "    return reserved\n" +
+        "end\n" +
+        "local reservedIndex = 1\n" +
+        "local nextLimit = tonumber($limit$)\n" +
+        "local nextOffset = 0\n" +
+        "while nextLimit > 0 do\n" +
+        "   local jobs = redis.call('ZRANGEBYSCORE', $readyQueue$, 0, $now$, 'WITHSCORES', 'LIMIT', nextOffset, nextLimit)\n" +
+        "   if next(jobs) == nil then\n" +
+        "       return reserved\n" +
+        "   end\n" +
+        "   for i=1,2*#jobs,2 do\n" +
+        "      local inRunningQueue = redis.call('ZSCORE', $runningQueue$, jobs[i])\n" +
+        "      if not inRunningQueue then\n" +
+        "          reserved[reservedIndex] = jobs[i]\n" +
+        "          reserved[reservedIndex + 1] = jobs[i + 1]\n" +
+        "          reservedIndex = reservedIndex + 2\n" +
+        "          redis.call('ZREM', $readyQueue$, jobs[i])\n" +
+        "          redis.call('ZADD', $runningQueue$, $ttl$, jobs[i])\n" +
+        "          nextLimit = nextLimit - 1\n" +
+        "      end\n" +
+        "   end\n" +
+        "   nextOffset = nextOffset + nextLimit\n" +
```

Comment:
there will be some overlap in your "pages", right (since you decrement nextLimit on line 66 and use it here as an offset on 69) ?